fs/mnemofs/mnemofs_journal.c

```diff
@@ -91,10 +91,33 @@
  * Private Types
  ****************************************************************************/
 
+/* NOTE: Even if higher level functions use path as struct mfs_path_s,
+ * journal only uses struct mfs_ctz_s to avoid problems during write and
+ * read of logs. The offsets in struct mfs_path_s will be applied by the
+ * search methods of higher functions.
+ */
+
+/* There is a byte-long hash that follows this on-flash. */
+
+struct jrnl_log_s
+{
+  mfs_t depth;
+  struct mfs_ctz_s new;
+  FAR struct mfs_ctz_s path[];
```

Comment:
C89 incompatible. Not sure if we are still expecting common code to be C89 compatible.

Reply:
Our CI doesn't try to catch any c89 compatible in common code. It's impossible to keep c89 combability without ci.